packages/react-integration/demo-app-ts/src/components/demos/HelperText/HelperTextDemo.tsx

```diff
@@ -0,0 +1,43 @@
+import { HelperText, HelperTextItem } from '@patternfly/react-core';
+import React from 'react';
+import InfoIcon from '@patternfly/react-icons/dist/js/icons/info-icon';
+import QuestionIcon from '@patternfly/react-icons/dist/js/icons/question-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+
+export class HelperTextDemo extends React.Component {
+  static displayName = 'HelperTextDemo';
+
+  componentDidMount() {
+    window.scrollTo(0, 0);
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <HelperText id="default-container">
+          <HelperTextItem id="default" icon={<InfoIcon />}>
+            This is default helper text
+          </HelperTextItem>
+        </HelperText>
+        <HelperText>
+          <HelperTextItem id="warning" variant="warning" icon={<ExclamationTriangleIcon />}>
+            This is warning helper text
+          </HelperTextItem>
+        </HelperText>
+        <HelperText>
+          <HelperTextItem id="success" isDynamic variant="success" icon={<CheckCircleIcon />}>
+            This is success helper text
+          </HelperTextItem>
+          <HelperTextItem id="invalid" isDynamic variant="invalid" icon={<ExclamationCircleIcon />}>
+            This is invalid helper text
+          </HelperTextItem>
+          <HelperTextItem id="indeterminate" isDynamic variant="indeterminate" icon={<QuestionIcon />}>
+            This is indeterminate helper text
+          </HelperTextItem>
+        </HelperText>
```

Comment:
Should we add an example and test for `isDynamic` and usage of `component`?

Reply:
There is an item with `isDynamic` already, unless you mean something else? I'm a little confused on what that flag is used for in the react side. 

I'll add a `component` example to the demo.

Reply:
Oh sorry I was looking at the cypress tests and saw we aren't looking for the dynamic class or list elements using `component`. 

> I'm a little confused on what that flag is used for in the react side

What that's supposed to do is allow the helper text item to switch states (warning, error, etc) as the field its associated with is being validated - which will update the icon, icon color, and potentially the font weight if the state changes between error and not-error. We have an issue to build a [login password strength demo](https://github.com/patternfly/patternfly/issues/4018) that would show how that works - here's the design for that, which shows how those elements would update on the fly as a field is being validated - https://marvelapp.com/prototype/6jehce7/screen/78511266

Reply:
Hmm I think I get it. I'm just a bit confused because the helper text component itself is stateless, its styling is determined by its props. The app would change an app-side state to update those props in a demo like that. In which case, does `isDynamic` need to be an exposed flag, or would it be something that always exists on a helper text item. Is there a downside to applying that modifier all the time?

Reply:
> Is there a downside to applying that modifier all the time?

Yeah, the helper text has 2 basic types so far. First is the solid color/regular state. In that state, it's just helper text that you would use associated with a field of some sort, and it may or may not change depending on the state of the field. For example, it could be normal/black text helper text that's used to describe a field. That text may or may not be replaced with error/red text if the field is invalid.

The other state is `isDynamic` which is an indication that it's a live type of helper text that will update as you update the field. It has grey text (or black when error/invalid) and a state colored icon and the icon and text changes as the field is validated.

I could see allowing the user to control the states/text that's displayed, but I could also see allowing the user to pass an object to the helper text item with the default, warning, error, and success state text to display, and as the helper text item `variant` is updated, we display the correct text or something like that?

cc @mcarrano is this inline with how you see this used?

Reply:
Yes, I think so.

Reply:
Ah I understand, thanks for explaining :)